R/LSTdescriptives.R

```diff
@@ -302,31 +300,63 @@ LSTdescriptives <- function(jaspResults, dataset, options, state = NULL) {
   nearestPointIndex <- which.min(abs(valueSet - targetValue))
   return(nearestPointIndex)
 }
-.descExplanationCT <- function(jaspResults, options) {
+
+.descExplanationCT <- function(jaspResults, options, colors) {
   jaspResults[["descExplanationCT"]] <- createJaspContainer(gettext("Explanation"))
   jaspResults[["descExplanationCT"]]$position <- 1
   jaspResults[["descExplanationCT"]]$dependOn(options = c("LSdescCentralOrSpread",
                                                           "LSdescCT",
                                                           "LSdescExplanationC"))
   
+  pdPlot <- createJaspPlot(title = gettext("Theoretical example distribution"), width = 700, height = 400)
+  pdPlot$position <- 1
+  
   mean <- 0
   sd <- 1
   skew <- 1000
+  distLimits <- c(mean - 4 * sd, mean + 4 * sd)
   
-  pdPlot <- createJaspPlot(title = gettext("Theoretical example distribution"), width = 700, height = 400)
-  pdPlot$position <- 1
+  data <- data.frame(x = .scaledSkewedNormal(100000, xi = mean, omega = sd, alpha = skew))
+  data <- subset(data, data$x > distLimits[1] & data$x < distLimits[2]) # remove values outside limits
+  
+  pdPlotObject <- .plotCTexampleDistribution(data)
+  
+  allCTs <- options[["LSdescCT"]] == "LSdescMMM"
+  
+  if (options[["LSdescCT"]] == "LSdescMedian"| allCTs) {
```

Comment:
```suggestion
  if (options[["LSdescCT"]] == "LSdescMedian" || allCTs) {
```
same for the 3 other if statements below